recipes-mender/artifacts/artifacts.bb

```diff
@@ -0,0 +1,50 @@
+DESCRIPTION = "Mender image artifact library"
+GO_IMPORT = "github.com/mendersoftware/artifacts"
+
+inherit go
+SRC_URI = "git://github.com/mendersoftware/artifacts.git;protocol=https"
+
+SRCREV = "${AUTOREV}"
+PVBASE := "${PV}"
+PV = "${PVBASE}+git${SRCPV}"
+
+LICENSE = "Apache-2.0"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=e3fc50a88d0a364313df4b21ef20c29e \
+                    file://LIC_FILES_CHKSUM.sha256;md5=ca079eb955443fa5828872744095fada \
+                    "
+S = "${WORKDIR}/git"
+B = "${WORKDIR}/build"
+
+GOPATHDIR = "${B}/src/${GO_IMPORT}"
+
+BBCLASSEXTEND = "native"
+
+FILES_${PN} += "${GOBIN_FINAL}/*"
+
+do_compile() {
+    # we could check out the sources at some destsuffix and use default
+    # do_compile from go.bbclass, but that would prevent us from always building
+    # the most recent version due to recursive expansion if SRCPV
+
+    install -d ${GOPATHDIR}
+    # we could also try symlinking ${S} into our fake GOPATH, however `go build...`
+    # ignores symlinks in GOPATH
+    rsync -av --exclude '.git' --delete ${S}/ ${GOPATHDIR}/
+
+    GOPATH=${B}:${STAGING_LIBDIR}/${TARGET_SYS}/go go env
+    if test -n "${GO_INSTALL}" ; then
+       GOPATH=${B}:${STAGING_LIBDIR}/${TARGET_SYS}/go go install -v ${GO_INSTALL}
+    fi
+}
+
+do_install_append() {
+    if test -e ${B}/pkg ; then
+        rsync -av --delete ${B}/pkg ${D}${GOROOT_FINAL}
+        chown -R root:root ${D}${GOROOT_FINAL}
+    fi
+    if test -e ${B}/bin ; then
+        install -d ${D}${bindir}
+        rsync -av --delete ${B}/bin/ ${D}${bindir}
+        chown -R root:root ${D}${GOROOT_FINAL}
```

Comment:
You can build it as `artifacts` (for target) and `artifacts-native`. `chown` is needed otherwise auto QA will raise host-user-contaminated warnings.